src/validate.rs

```diff
@@ -0,0 +1,24 @@
+//! Input buffer validation API (Multi-line editing)
+
+/// This trait provides an extension interface for determining whether
+/// the current input buffer is valid. Rustyline uses the method
+/// provided by this trait to decide whether hitting the enter key
+/// will end the current editing session and return the current line
+/// buffer to the caller of `Editor::readline` or variants.
+pub trait Validator {
+    /// Takes the currently edited `line` and returns a bool
+    /// indicating whether it is valid or not. The most common
+    /// validity check to implement is probably whether the input is
+    /// complete or not, for instance ensuring that all delimiters are
+    /// fully balanced.
+    ///
+    /// If you implement more complex validation checks it's probably
+    /// a good idea to also implement a `Hinter` to provide feedback
+    /// about what is invalid.
+    #[allow(unused_variables)]
+    fn is_valid(&self, line: &str) -> bool {
```

Comment:
I wonder if `is_valid` is the best name here. There could be cases where the line is syntactically invalid in a way that appending a more characters couldn’t fix. For example, this is terminally broken because the braces don’t match:

```rust
fn my_function(foo: i64]
```

Maybe `is_complete`, `is_closed`, `has_open_delimiter` could be better options?

Reply:
I'm also dissatisfied with the name.  I wanted to stay away from "complete" language since there is already a `Completer` helper trait with a `complete` function on it that is completely different in functionality.  The other names seem rather narrowly focused on the cases of closing delimiters but there's no real reason why that is the only usage for this.  Though it's probably a major one.

Reply:
How about something direct like `should_continue_line`?